src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectionTestWithSSLCert/CertificateTestWithTdsServer.cs

```diff
@@ -0,0 +1,160 @@
+{
+    public class CertificateTestWithTdsServer
+    {
+        private static readonly string s_fullPathToPowershellScript = Path.Combine(Directory.GetCurrentDirectory(), "makepfxcert.ps1");
+        private static readonly string s_fullPathToPfx = Path.Combine(Directory.GetCurrentDirectory(), "localhostcert.pfx");
+
+        public CertificateTestWithTdsServer()
+        {
+            CreatePfxCertificate(s_fullPathToPowershellScript);
+        }
+
+        [Theory]
+        [MemberData(nameof(ConnectionTestParametersData.GetConnectionTestParameters), MemberType = typeof(ConnectionTestParametersData))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void BeginWindowsConnectionTest(ConnectionTestParameters connectionTestParameters)
+        {
+            ConnectionTest(connectionTestParameters);
+        }
+
+        [Theory]
+        [MemberData(nameof(ConnectionTestParametersData.GetConnectionTestParameters), MemberType = typeof(ConnectionTestParametersData))]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public void BeginLinuxConnectionTest(ConnectionTestParameters connectionTestParameters)
+        {
+            ConnectionTest(connectionTestParameters);
+        }
+
+        private void ConnectionTest(ConnectionTestParameters connectionTestParameters)
+        {
+            string userId = string.Empty;
+            string password = string.Empty;
+            SqlConnectionStringBuilder builder = new(DataTestUtility.TCPConnectionString);
+            userId = builder.UserID;
+            password = builder.Password;
+
+            using TestTdsServer server = TestTdsServer.StartTestServer(enableFedAuth: false, enableLog: false, connectionTimeout: 15,
+                methodName: "", new X509Certificate2(s_fullPathToPfx, "nopassword", X509KeyStorageFlags.UserKeySet),
+                encryptionType: connectionTestParameters.TdsEncryptionType);
+
+            if (userId != string.Empty)
+            {
+                builder = new(server.ConnectionString)
+                {
+                    UserID = userId,
+                    Password = password,
+                    TrustServerCertificate = connectionTestParameters.TrustServerCertificate,
+                    Encrypt = connectionTestParameters.Encrypt,
+                };
+            }
+            else
+            {
+                builder = new(server.ConnectionString)
+                {
+                    IntegratedSecurity = true,
+                    TrustServerCertificate = connectionTestParameters.TrustServerCertificate,
+                    Encrypt = connectionTestParameters.Encrypt,
+                };
+            }
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && userId == string.Empty)
+            {
+                builder.IntegratedSecurity = false;
+                builder.UserID = "user";
+                builder.Password = "password";
+            }
```

Comment:
This seems strange. Why do we switch between integrated and un/pw here? And why use the un/pw from the config? Doesn't the auth not matter since TestTdsServer doesn't care about auth here? Seems like this could be simplified to just a few lines... No?
```suggestion
            using TestTdsServer server = TestTdsServer.StartTestServer(enableFedAuth: false, enableLog: false, connectionTimeout: 15,
                methodName: "", new X509Certificate2(s_fullPathToPfx, "nopassword", X509KeyStorageFlags.UserKeySet),
                encryptionType: connectionTestParameters.TdsEncryptionType);

            SqlConnectionStringBuilder builder = new(server.ConnectionString)
            {
                TrustServerCertificate = connectionTestParameters.TrustServerCertificate,
                Encrypt = connectionTestParameters.Encrypt,
            };
```

Reply:
This was addressed in PR #2487.